libs/coin-evm/src/transaction.ts

```diff
@@ -118,22 +129,63 @@ export const toTransactionRaw = (tx: EvmTransaction): EvmTransactionRaw => {
     txRaw.additionalFees = tx.additionalFees.toFixed();
   }
 
+  if (tx.nft) {
+    txRaw.nft = {
+      tokenId: tx.nft.tokenId,
+      contract: tx.nft.contract,
+      quantity: tx.nft.quantity.toFixed(),
+      collectionName: tx.nft.collectionName,
+    };
+  }
+
   return txRaw as EvmTransactionRaw;
 };
 
 /**
  * Returns the data necessary to execute smart contracts.
  * As of now, only used to create ERC20 transfers' data
  */
-export const getTransactionData = (transaction: EvmTransaction): Buffer | undefined => {
-  const contract = new ethers.utils.Interface(ERC20ABI);
-  const data = contract.encodeFunctionData("transfer", [
-    transaction.recipient,
-    transaction.amount.toFixed(),
-  ]);
-
-  // removing 0x prefix
-  return Buffer.from(data.slice(2), "hex");
+export const getTransactionData = (
+  account: Account,
+  transaction: EvmTransaction,
+): Buffer | undefined => {
+  switch (transaction.mode) {
+    case "send": {
+      const contract = new ethers.utils.Interface(ERC20ABI);
+      const data = contract.encodeFunctionData("transfer", [
+        transaction.recipient,
+        transaction.amount.toFixed(),
+      ]);
+
+      // removing 0x prefix
```

Comment:
Maybe add why we need to do this

Reply:
Don't think that's necessary here, removing the 0x prefix is like a 101 thing when you work with Buffers and hexadecimals, so I don't see what I could really add to this 🤷‍♂️